### ISSUE TITLE
Attachment content not being updated

### ISSUE THREAD

**Issue by Real-Gecko** [viczem/ansible-keepass#54]
I've added an attachment to my file with certain content:
I read it like this:
```ansible
    - name: Backend postgres vars
      ansible.builtin.include_vars:
        file: "{{ lookup('viczem.keepass.keepass', 'infrastructure/infisical', 'attachments', 'infisical-postgres.yml') }}"
        name: infisical_postgres_env
```
However I've noticed that whenever I change the attachment in DB, this plugin still reads old value on next run.

**Comment by hgtok** [comment]
there is a TTL logic to reuse the open socket. could it be reusing the old connection?

**Comment by Real-Gecko** [comment]
I tried two subsequent runs with `-vvvv` option and in both cases I see:
```log
Loading collection viczem.keepass from /home/realgecko/.ansible/collections/ansible_collections/viczem/keepass
KeePass: run socket for mydb.kdbx
KeePass: try connect to socket 0/10
KeePass: try connect to socket 1/10
KeePass: send password to '/tmp/ansible-keepass-7acc9cfc.sock'
KeePass: open socket for mydb.kdbx -> /tmp/ansible-keepass-7acc9cfc.sock
KeePass: connect to '/tmp/ansible-keepass-7acc9cfc.sock'
KeePass: fetch ['infrastructure/infisical', 'attachments', 'infisical-postgres.yml']
KeePass: disconnect from '/tmp/ansible-keepass-7acc9cfc.sock'
```
Socket path in both cases is the same `/tmp/ansible-keepass-7acc9cfc.sock` can it be related?

**Comment by Real-Gecko** [comment]
Tried settings `keepass_ttl` to 1, does not help

**Comment by Real-Gecko** [comment]
Tried closing and reopening DB: same result
Tried deleting record history: same result
Trying to pinpoint when update happens, no luck yet

**Comment by Real-Gecko** [comment]
Also when I try it like this:
```yml
    - name: Infisical
      community.docker.docker_swarm_service:
        name: infisical
        image: infisical/infisical:v0.77.3-postgres
        env_files:
          - "{{ lookup('viczem.keepass.keepass', 'infrastructure/infisical', 'attachments', 'infisical-postgres.yml') }}"
```
I get an error `"Error parsing module parameters: [Errno 2] No such file or directory: '/tmp/tmp37kpdszh.infisical-postgres.yml'"`

**Comment by Real-Gecko** [comment]
File exists however, and I also noticed that files from previous runs exist too, so not removed after being read:
![image](https://github.com/user-attachments/assets/5660c413-5eb3-40e1-92e1-3fcbe3ad3183)

**Comment by Real-Gecko** [comment]
Same happens to custom properties BTW, I've added custom property to the entry, but `lookup('viczem.keepass.keepass', 'infrastructure/infisical', 'custom_properties', 'env_vars')` says that it cannot find such property.

**Comment by Real-Gecko** [comment]
Ok, some progress:
1. Initially launched with `keepass_ttl` not being set the module creates keepass unix socket and keeps it open forever
2. After system reboot (and obviously clearing the socket) and settings `keepass_ttl` to `1` everything works as expected:
 - socket is created
 - data read
 - after 1 second socket closed
 - on next run cycle repeats and updated secrets are read
 - after socket is closed, temporary files removed

So, there're two problems I see:
1. If `keepass_ttl` initially is not set, then updating the value in playbook does not take new value into account, so socket is never closed and no new data is read
2. For some reason opened socket does not provide updated data as if initial opening of the socket loads all data into memory and no updates from disk are read.

**Comment by viczem** [comment]
Please see README: How it works. And default socket ttl is 60 seconds

**Comment by Real-Gecko** [comment]
> Please see README: How it works. And default socket ttl is 60 seconds

I think it did not work, because socket remained open until reboot.

**Comment by viczem** [comment]
I just tested and found no problems with TTL (default is 60 seconds). You can also close socket by https://github.com/viczem/ansible-keepass/blob/b633a57fc316c86f5b0c2b3b7afcd5eccdde4a62/docs/examples/example-playbook.yml#L38

**Comment by Real-Gecko** [comment]
> I just tested and found no problems with TTL (default is 60 seconds). You can also close socket by
> 
> https://github.com/viczem/ansible-keepass/blob/b633a57fc316c86f5b0c2b3b7afcd5eccdde4a62/docs/examples/example-playbook.yml#L38

Is this socket behavior comes from `pykeepass`?

**Comment by viczem** [comment]
The socket implemented by the plugin. The `pykeepass` is for parsing kdbx files.
To make sure that the socket has not closed and continues to work after the TTL expires, try running the command `ps -ef | grep ansible`. The output should like
 `.../.ansible/collections/ansible_collections/viczem/keepass/plugins/lookup/keepass.py ...ansible-keepass/docs/examples/example.kdbx /tmp/ansible-keepass-20d8af1e.sock 3`

The last number is the TTL in seconds. What number in your case ?

**Comment by Real-Gecko** [comment]
Ok, I see, ttl value on socket was 60, I guess I was doing changes and rerunning ansible too quiclky, without reaching 60s limit and thus was getting old values from db. Maybe this can be reflected in documentation more thorougly? And also maybe we can have some option to work without socket, reading files directly?

**Comment by viczem** [comment]
To clarify, TTL starts counting from the moment of the last access to the socket. In other words, we specify the idle time after which the socket will close automatically. If the task in the playbook is so long that it exceeds the TTL, the socket will be closed and when Ansible accesses the plugin, a new socket will be created.

Without a socket, there will be many operations with kdbx file decryption (essentially as many times as Ansible will read the variable). But through a socket the decryption once at the start.

This plugin is for read-only kdbx files. It is assumed that you will use a kdbx file without need to change anything in it during running Ansible process.

**Comment by Real-Gecko** [comment]
So, socket connection copies DB into memory and does not follow changes on disk?

**Comment by viczem** [comment]
Yes, in memory decrypted data of your kdbx file, while socket is open

**Comment by Real-Gecko** [comment]
> Also when I try it like this:
> 
> ```yaml
>     - name: Infisical
>       community.docker.docker_swarm_service:
>         name: infisical
>         image: infisical/infisical:v0.77.3-postgres
>         env_files:
>           - "{{ lookup('viczem.keepass.keepass', 'infrastructure/infisical', 'attachments', 'infisical-postgres.yml') }}"
> ```
> 
> I get an error `"Error parsing module parameters: [Errno 2] No such file or directory: '/tmp/tmp37kpdszh.infisical-postgres.yml'"`

What about this one?

**Comment by Real-Gecko** [comment]
maybe something related to socket also?

**Comment by viczem** [comment]
When socket close it will delete all related files. If a playbook has not yet completed but the socket was closed, then accessing the file cannot find it.

**Comment by viczem** [comment]
Have you tried getting the attachment file this way?

```
- name: "Export file: attachment.txt"
    viczem.keepass.attachment:
      database: "{{ keepass_dbx }}"
      password: "{{ keepass_psw }}"
      entrypath: example/attachments
      attachment: "attachment.txt"
      dest: "{{ keepass_attachment_1_name }}"
```

**Comment by Real-Gecko** [comment]
> When socket close it will delete all related files. If a playbook has not yet completed but the socket was closed, then accessing the file cannot find it.

Well it happened with TTL 60 when I was doing things fast, so socket was definitely open, and also file existed in filesystem :D

**Comment by Real-Gecko** [comment]
> Have you tried getting the attachment file this way?
> 
> ```
> - name: "Export file: attachment.txt"
>     viczem.keepass.attachment:
>       database: "{{ keepass_dbx }}"
>       password: "{{ keepass_psw }}"
>       entrypath: example/attachments
>       attachment: "attachment.txt"
>       dest: "{{ keepass_attachment_1_name }}"
> ```

Nope, only with lookup `{{ lookup('viczem.keepass.keepass', 'infrastructure/infisical', 'attachments', 'infisical-postgres.yml') }}`

**Comment by Real-Gecko** [comment]
> > When socket close it will delete all related files. If a playbook has not yet completed but the socket was closed, then accessing the file cannot find it.
> 
> Well, it happened with TTL=60 when I was doing things fast, so socket was still open when error occured, and also file existed in filesystem :D
